lte/gateway/c/core/oai/tasks/amf/Registration.cpp

```diff
@@ -220,18 +220,32 @@ int amf_proc_registration_request(amf_ue_ngap_id_t ue_id,
     }
   }
 
+  /* Implicit deregistartion of existing context should be triggered if
+   * same TMSI used and trigger fresh registration request*/
+  if (ies->guti) {
+    amf_app_desc_t* amf_app_desc_p = get_amf_nas_state(false);
+    ue_m5gmm_context_s* guti_ue_mm_ctx = NULL;
+
+    guti_ue_mm_ctx =
+        amf_ue_context_exists_guti(&amf_app_desc_p->amf_ue_contexts, ies->guti);
+    if (guti_ue_mm_ctx) {
+      old_ue_id = guti_ue_mm_ctx->amf_ue_ngap_id;
```

Comment:
@Sathyaj27, 
        Is a new context creation required if already UE Context is available, can amf reuse the same context.?
        AMF can avoid interaction with Subscribed for  Subscriber profile for Mobility Management and Session Management.

Reply:
New registration is coming with new gnb id, Hence requires new context. The older context hence needs to be deleted.